src/runtime/pkg/govmm/qemu/qemu.go

```diff
@@ -377,6 +382,14 @@ func (object Object) QemuParams(config *Config) []string {
 
 	case TDXGuest:
 		objectParams = append(objectParams, prepareObjectWithTdxQgs(object))
+		objectParams = append(objectParams, string(object.Type))
+		objectParams = append(objectParams, fmt.Sprintf("id=%s", object.ID))
+		if object.Debug {
+			objectParams = append(objectParams, "debug=on")
+		}
+		if len(object.InitdataDigest) > 0 {
+			objectParams = append(objectParams, fmt.Sprintf("mrconfigid=%s", object.InitdataDigest))
+		}
```

Comment:
Ding, this is **NOT** the way to go.
What we want to do is actually build this as part of the `prepareObjectWithTdxQgs`, as the TDX object must use the qom-qapi json notation.

So, please, take a look here: https://github.com/kata-containers/kata-containers/blob/29ce2205a1163a1b3b0ca11a8e310d3a512a344b/src/runtime/pkg/govmm/qemu/qemu.go#L466

You'll need to add `mrconfigid` to https://github.com/kata-containers/kata-containers/blob/29ce2205a1163a1b3b0ca11a8e310d3a512a344b/src/runtime/pkg/govmm/qemu/qemu.go#L437-L442 (and while doing it, please, go ahead and also add `mrowner` and `mrownerconfig`).

And also to here: https://github.com/kata-containers/kata-containers/blob/29ce2205a1163a1b3b0ca11a8e310d3a512a344b/src/runtime/pkg/govmm/qemu/qemu.go#L322

Here's a non-tested short diff that can make your life easier, hopefully:
```
diff --git a/src/runtime/pkg/govmm/qemu/qemu.go b/src/runtime/pkg/govmm/qemu/qemu.go
index 47322c8032..94d66a7cb2 100644
--- a/src/runtime/pkg/govmm/qemu/qemu.go
+++ b/src/runtime/pkg/govmm/qemu/qemu.go
@@ -320,6 +320,20 @@ type Object struct {
 
        // QgsPort defines Intel Quote Generation Service port exposed from the host
        QgsPort uint32
+
+       // MrConfigId defines the ID for non-owner-defined configuration of the TDX guest,
+       // like run-time or OS configuration.
+       // The expected format is base64 encoded SHA384 digest
+       MrConfigId string
+
+       // MrOwner defines the ID for the TDX guest owner
+       // The expected format is base64 encoded SHA384 digest
+       MrOwner string
+
+       // MrOwnerConfig defines the ID for owner-defined configuration of the TDX guest,
+       // like the workload rather than the run-time or OS.
+       // The expected format is base64 encoded SHA384 digest
+       MrOwnerConfig string
 }
 
 // Valid returns true if the Object structure is valid and complete.
@@ -437,6 +451,9 @@ type SocketAddress struct {
 type TdxQomObject struct {
        QomType               string        `json:"qom-type"`
        Id                    string        `json:"id"`
+       MrConfigId            string        `json:"mrconfigid"`
+       MrOwner               string        `json:"mrowner"`
+       MrOwnerConfig         string        `json:"mrownerconfig"`
        QuoteGenerationSocket SocketAddress `json:"quote-generation-socket"`
        Debug                 *bool         `json:"debug,omitempty"`
 }
@@ -465,7 +482,7 @@ func (this *TdxQomObject) String() string {
 
 func prepareObjectWithTdxQgs(object Object) string {
        qgsSocket := SocketAddress{"vsock", fmt.Sprint(VsockHostCid), fmt.Sprint(object.QgsPort)}
-       tdxObject := TdxQomObject{string(object.Type), object.ID, qgsSocket, nil}
+       tdxObject := TdxQomObject{string(object.Type), object.ID, object.MrConfigId, object.MrOwner, object.MrOwnerConfig, qgsSocket, nil}
 
        if object.Debug {
                *tdxObject.Debug = true
```